sentence_transformers/SentenceTransformer.py

```diff
@@ -111,6 +112,15 @@ def __init__(
         if cache_folder is None:
             cache_folder = os.getenv("SENTENCE_TRANSFORMERS_HOME")
 
+        if device is None:
+            device = get_device_name()
+            logger.info("Use pytorch device_name: {}".format(device))
+
+        if device == "hpu":
```

Comment:
Can we be sure that `optimum-habana` is installed here?

Reply:
good suggestion. i will add the check. thanks.